pymc_extras/statespace/models/structural/components/cycle.py

```diff
@@ -0,0 +1,307 @@
+
+    def __init__(
+        self,
+        name: str | None = None,
+        cycle_length: int | None = None,
+        estimate_cycle_length: bool = False,
+        dampen: bool = False,
+        innovations: bool = True,
+        observed_state_names: list[str] | None = None,
+    ):
+        if observed_state_names is None:
+            observed_state_names = ["data"]
+
+        if cycle_length is None and not estimate_cycle_length:
+            raise ValueError("Must specify cycle_length if estimate_cycle_length is False")
+        if cycle_length is not None and estimate_cycle_length:
+            raise ValueError("Cannot specify cycle_length if estimate_cycle_length is True")
+        if name is None:
+            cycle = int(cycle_length) if cycle_length is not None else "Estimate"
+            name = f"Cycle[s={cycle}, dampen={dampen}, innovations={innovations}]"
+
+        self.estimate_cycle_length = estimate_cycle_length
+        self.cycle_length = cycle_length
+        self.innovations = innovations
+        self.dampen = dampen
+        self.n_coefs = 1
+
+        k_endog = len(observed_state_names)
+
+        k_states = 2 * k_endog
+        k_posdef = 2 * k_endog
+
+        obs_state_idx = np.zeros(k_states)
+        obs_state_idx[slice(0, k_states, 2)] = 1
+
+        super().__init__(
+            name=name,
+            k_endog=k_endog,
+            k_states=k_states,
+            k_posdef=k_posdef,
+            measurement_error=False,
+            combine_hidden_states=True,
+            obs_state_idxs=obs_state_idx,
+            observed_state_names=observed_state_names,
+        )
+
+    def make_symbolic_graph(self) -> None:
+        if self.k_endog == 1:
+            self.ssm["design", 0, slice(0, self.k_states, 2)] = 1
+            self.ssm["selection", :, :] = np.eye(self.k_states)
+            init_state = self.make_and_register_variable(f"{self.name}", shape=(self.k_states,))
+
+        else:
+            Z = np.array([1.0, 0.0]).reshape((1, -1))
+            design_matrix = linalg.block_diag(*[Z for _ in range(self.k_endog)])
+            self.ssm["design", :, :] = pt.as_tensor_variable(design_matrix)
+
+            R = np.eye(2)  # 2x2 identity for each cycle component
+            selection_matrix = linalg.block_diag(*[R for _ in range(self.k_endog)])
+            self.ssm["selection", :, :] = pt.as_tensor_variable(selection_matrix)
+
+            init_state = self.make_and_register_variable(f"{self.name}", shape=(self.k_endog, 2))
+
+        self.ssm["initial_state", :] = init_state.ravel()
+
+        if self.estimate_cycle_length:
+            lamb = self.make_and_register_variable(f"{self.name}_length", shape=())
+        else:
+            lamb = self.cycle_length
+
+        if self.dampen:
+            rho = self.make_and_register_variable(f"{self.name}_dampening_factor", shape=())
+        else:
+            rho = 1
+
+        T = rho * _frequency_transition_block(lamb, j=1)
+        if self.k_endog == 1:
```

Comment:
No need for a special case here, `block_diag([x]) = x` and we rewrite away the useless `block_diag`